Landing page

ManuelRemer opened the pull request:
Implemented landing page, without any functionality right now. Styled it, tried to make it responsive at least for mobile devices. Does not work for rotated view, any ideas?

namirsab commented on src/App.css:
```diff
@@ -6,8 +6,8 @@
 }
 
-.App {
+/* .App {
```
Please remove unused code or commented code

namirsab commented on src/components/LandingPage.css:
```diff
@@ -0,0 +1,58 @@
+  background-color: var(--color-sand);
+  display: grid;
+  grid-template-rows: [header]15vh [image]35vh [rest]50vh;
+  grid-template-columns: 1fr;
```
You don't need to define `grid-template-columns` if you only have one column.

ManuelRemer replied:
Ok, I did so before but then I had kind of gutters on the left and right. But I`ll try again.

namirsab commented on src/components/LandingPage.css:
```diff
@@ -0,0 +1,58 @@
+  width: 100vw;
+}
+.landing-header {
+  grid-row: [header];
```
you don't need to wrap `header` with `[]`. When defining a `grid-row` using a line name, you just use the name
Also you need to define the beginning and the end like this.
`grid-row: header / image`

check the gridrow property in mdn:
https://developer.mozilla.org/en-US/docs/Web/CSS/grid-row

namirsab replied:
You have this problem in several places, check them as well

namirsab commented on src/components/LandingPage.css:
```diff
@@ -0,0 +1,58 @@
+  grid-template-columns: 1fr;
+  justify-content: center;
+  height: 100vh;
+  width: 100vw;
```
Elements by default will try to get the whole width. I don't think you need to set the `width`here.

namirsab commented on src/components/LandingPage.js:
```diff
@@ -0,0 +1,28 @@
+import "./LandingPage.css";
+import StandardButton from "./StandardButton";
+import LandingBread from "../images/LandingBread.jpg";
```
You shouldn't call this `LandingBread` as it looks like it would be a component. It's just the imageSrc, so I'd call this `landingBreadSrc` or something similar

namirsab commented on src/components/LandingPage.js:
```diff
@@ -0,0 +1,28 @@
+        <div className="landing-text">
+          <p>Design your own</p>
+          <p className="landing-text-highlight">Bread</p>
+          <p>with the ingrediants</p> <p>you want</p>
```
Little typo here : `ingrediants` instead of `ingredients`

namirsab commented on src/components/StandardButton.css:
```diff
@@ -0,0 +1,13 @@
+.standard-button {
+  margin: 5px;
+  /* border: 3px solid var(--color-fonts); */
```
Let's remove commented code :)

namirsab commented on src/index.js:
```diff
@@ -1,6 +1,7 @@
 import React from "react";
 import ReactDOM from "react-dom";
 import "./index.css";
+import "./color.css";
```
It seems you are importing `color.css` but it's an empty file. You are setting the colors now in the `index.css`
Either define the css variables for the colors in the `colors.css` file and use it, or remove this import.
I'd go for having the colors in the `colors.css` file though.

namirsab reviewed (changes requested):
Hey @ManuelRemer 
Good job, you've set your first page of your app!
I've added some comments with some changes. Please take a look

namirsab commented on src/components/LandingPage.css:
```diff
@@ -0,0 +1,58 @@
+.landing-page {
+  background-color: var(--color-sand);
+  display: grid;
+  grid-template-rows: [header]15vh [image]35vh [rest]50vh;
```
I think the problem of your app not "working" well in landscape mode is because your content is bigger than the view port. And you have set the `rest` row to have `50vh`. But your content is bigger than the viewport height, hence you see that white stripe below.

What you can do is leave the `rest` to auto
```css
grid-template-rows: [header]15vh [image]35vh [rest]auto;
```

marwinburesch commented on src/components/StandardButton.js:
```diff
@@ -0,0 +1,4 @@
+import "./StandardButton.css";
+export default function StandardButton({ name }) {
+  return <div className="standard-button">{name}</div>;
```
It's not a `<button>` though :D

marwinburesch commented on src/components/StandardButton.css:
```diff
@@ -0,0 +1,13 @@
+  /* border: 3px solid var(--color-fonts); */
+  border-radius: 50px;
+  box-shadow: 5px 5px 13px -1px rgba(0, 0, 0, 0.22);
+  width: 200px;
```
The width should be handled somewhere else and not be restricted like this. Maybe a relative unit or with one of the layout options we have at hand in CSS. _OR_ just by the content in the button

marwinburesch reviewed (changes requested):
It's coming along! Good job 🐜

ManuelRemer commented:
> Hey @ManuelRemer
> Good job, you've set your first page of your app!
> I've added some comments with some changes. Please take a look



> Hey @ManuelRemer
> Good job, you've set your first page of your app!
> I've added some comments with some changes. Please take a look

Thanx, i will take care of the comments conscientiously
